Realm/Realm/Realm.csproj

```diff
@@ -1,4 +1,50 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <UsingTask
+    TaskName="ExtractChangelog"
+    TaskFactory="RoslynCodeTaskFactory"
+    AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+
+    <ParameterGroup>
+        <FilePath ParameterType="System.String" Required="true" />
+        <ExtractedText ParameterType="System.String" Output="true" />
+    </ParameterGroup>
+
+    <Task>
+      <Using Namespace="System"/>
+      <Using Namespace="System.IO"/>
+      <Using Namespace="System.Text.RegularExpressions"/>
+      <Code Type="Fragment" Language="cs">
+<![CDATA[
+try
+{
+    var textToParse = File.ReadAllText(FilePath);
+    var regex = new Regex("(?sm)^(## \\d{1,2}\\.\\d{1,2}\\.\\d{1,2}(?:-beta\\.\\d{1,2})? \\(\\d{4}-\\d{2}-\\d{2}\\))(.+?)(?=\n## \\d{1,2}\\.\\d{1,2}\\.\\d{1,2}(?:-beta\\.\\d{1,2})? \\(\\d{4}-\\d{2}-\\d{2}\\))");
```

Comment:
This regex is slightly different from the one in `extract-release-notes`. Can you modify it to handle other prerelease types, such as `-alpha.2`, `-pre.123`, or `-flexsync.1`.